### ISSUE TITLE
Saving a connectioninfo when its required credential repo is unavailable should not overwrite the associated credential

### ISSUE THREAD

**Issue by sparerd** [mRemoteNG/mRemoteNG#659]
*This is only relevant for credential manager work being developed under #208*

Currently, every ConnectionInfo holds a direct reference to its associated ICredentialRecord. During serialization, every ConnectionInfo saves the id of its associated `cred`. An example snippet looks something like this (most of the properties have been deleted from this entry to better highlight the relationship):
```xml
<Node Name="New Connection" Id="d9733e7f-085c-4d77-95cc-3bd625e1cdf4" CredentialId="cbbc07dc-4ec3-4cce-9b03-b5f2d001b94e" />
```
In that sample, the credential with id `cbbc07dc-4ec3-4cce-9b03-b5f2d001b94e` is bound to connection with id `d9733e7f-085c-4d77-95cc-3bd625e1cdf4`. When the connection is loaded, mRemoteNG will search its credential system for a `cred` with the matching id.

Because this binding happens during the deserialization of the confCons file, issues crop up if no credential exists that matches the required id. If we load con1 which requires cred1 but cred1 isn't available, then when we save the list of connections, con1 won't have an association to save (in the statement `con1.CredentialRecord.Id`, `CredentialRecord` would be null).

ConnectionInfo should not save a direct reference to an ICredentialRecord but instead just save the cred record ID. When the system needs to retrieve user/pass/domain info, it should request this from the credential manager. This will allow for dynamically loading/unloading credential repos without severing con/cred associations.

**Comment by sparerd** [comment]
ConnectionInfos now only know about credential guids.

Need to fix how credentials are displayed in the config window:
 - [ ] if a credential id exists, the credential should be provided to the config window to display as a complex object (users can modify the credential from the config window)
- [ ] if a credential id does not exist for a connection info, display a message that no credential is associated and provide a drop down to select from the set of currently loaded credentials

**Comment by sparerd** [comment]
Fix serialization:
- [ ] the `CredentialId` field is being serialized with "mRemoteNG.Tools.Maybe`1[System.Guid]" instead of the guid (or empty string)
- [ ] deserializing a guid does not seem to create the correct Maybe<Guid>. Or it might be getting corrupted at some point.

**Comment by sparerd** [comment]
To get around some reflection issues in the FilteredPropertyGrid, each connection still has an ICredentialRecord reference. However, these references are proxied from the credential manager and never saved in a connection record directly. This will be a little confusing from the API side of things, since connections also have a CredentialRecordId property that is used for serialization. At some point this should get cleaned up, but we will need to ditch the reflection-based property window before that happens.

**Comment by sparerd** [comment]
This enhancement is resolved and merged into `develop` in commit 1c7592c70711f576ac237e985ac6e18d9d4ff261
